### PR TITLE
Use a python script for untarring qnx

### PR DESCRIPTION
**Pull request by Hoverbear**
We have thusfar held off testing QNX on Windows since untarring it on CircleCI's Windows runners takes.... over 43 minutes. We're hoping moving to Python will avoid that as it did in other places.

If this *doesn't* help, then I guess CircleCI just doesn't provide Windows Runners with enough IOPS for this to make sense.

### CODE REVIEW

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try (again)

**Comment by Hoverbear** [comment]
I guess this is no longer valid https://discuss.circleci.com/t/activate-python-virtualenv-for-whole-job/14434/2

**Comment by Hoverbear** [comment]
bors try-

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
I replaced the deployment on AWS with a new one that was built with `--dereference`

**Comment by Hoverbear** [comment]
bors try

**Comment by Hoverbear** [comment]
bors try

File: ferrocene/ci/scripts/setup-qnx-toolchain.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import argparse
+import tarfile
+import boto3
+import zstandard
+
+TARBALL_BUCKET = "ferrocene-ci-mirrors"
+TARBALL_KEY = "manual/qnx/qnx710-deployment.tar.xz"
+
+def main():
```
Comment (Hoverbear) [inline ferrocene/ci/scripts/setup-qnx-toolchain.py:16]: Next time I gotta do tarball stuff on Windows I';m gonna make this script more generic and reuse it. For now, I'd just like to get the QNX self-test working.

File: .circleci/calculate-parameters.py [hunk 2]
```diff
@@ -97,6 +97,12 @@ def calculate_docker_image_tag(platform_plus_image: str):
             hash.update(file.encode("utf-8"))
             hash.update(f.read())
 
+    # The docker files depend on `requirements.txt` for their venv.
+    requirements_file = "requirements.txt"
+    with open(requirements_file, "rb") as rf:
+        hash.update(requirements_file.encode("utf-8"))
+        hash.update(rf.read())
```
Comment (pietroalbini) [inline .circleci/calculate-parameters.py:104]: Can't you add it to `all_files`?

Comment (Hoverbear) [reply .circleci/calculate-parameters.py]: Hm, yes, that is indeed more clever. 🤔

**Review by pietroalbini — APPROVED** [review]
bors merge

**Pull request merged**